### ISSUE TITLE
OAuth 2.0 Client Credentials Flow Failing

### ISSUE THREAD

**Issue by giometrix** [jenius-apps/nightingale-rest-api-client#9]
From my reddit post:

> I just tried this out... I really like the idea for the auth feature ; but for some reason I keep getting a restsharp exception (invalid scope) . My company uses standard oath 2.0 using identity server 3; so it should be working .

> I tried to see what’s being sent across the wire , but for some reason the request does not get sent when I open fiddler (admittedly I didn’t look too far into it ).

Here is a screenshot from Postman:

![image](https://user-images.githubusercontent.com/13258728/60749784-20ad7100-9f6d-11e9-9343-1031ae76bdda.png)

And from Nightingdale:

![image](https://user-images.githubusercontent.com/13258728/60749799-4fc3e280-9f6d-11e9-973a-c131b8b31ea8.png)

Here is the error text:

```
{"error":"invalid_scope"}

   at Nightingale.HttpClients.RestSharpClient.RestSharpAuthenticator.<GetAccessToken>d__2.MoveNext() + 0x353
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x5c
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x44
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x1c
   at Nightingale.HttpClients.RestSharpClient.RestSharpAuthenticator.<GetOAuth2Token>d__1.MoveNext() + 0x214
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x5c
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x44
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x1c
   at Nightingale.ViewModels.AuthControlViewModel.<FetchToken>d__73.MoveNext() + 0xd6
```

**Comment by dpaulino** [comment]
Hey there! Haven't seen this before. Looks like a regression when we launched v2. Thanks for reporting! Investigating this asap.

**Comment by dpaulino** [comment]
Found the bug. The scope was not properly being added into the authentication request for the client credential flow. Thanks for spotting this! Just fixed this now and it'll be available in the next public release. If you're interested in trying out the fix sooner, feel free to sign up for the beta [here](https://forms.microsoft.com/Pages/ResponsePage.aspx?id=DQSIkWdsW0yxEjajBLZtrQAAAAAAAAAAAANAAQduPxpURU8zSUFNUTBYRDVSUkhJS0JQOE8yTFZYTy4u). 

Closing ticket for now and I will comment again when the update is released with this fix. Thanks again!

**Comment by dpaulino** [comment]
@giometrix the fix is now available to preview members as v2.4.1. In case you signed up, let me know if you're encountering the issue still or if you see new/different problem. You may need to head to [Nightingale's store page](https://www.microsoft.com/en-us/p/nightingale-rest-api-client/9n2t6f9f5zdn) to force Windows to check for an update.
